src/pages/MainQuizScreen.tsx

```diff
@@ -93,19 +50,64 @@ const MainQuizScreen: React.FC = () => {
 
   const navigate = useNavigate();
 
+  const dispatch = useAppDispatch();
+
   const handleConfirmModal = () => {
     navigate(paths.home);
   };
 
   const finishQuiz = () => {
+    dispatch(setEndTime(Date.now()));
     setTimeout(() => navigate(paths.results), 1500);
   };
 
-  const [question, answers]: [Question | undefined, Answer[]] = useMemo(() => {
-    const _question = quizQuestions.find((q) => q.id === currentQuestionId);
-    const _answers = getAnswersByQuestionId(currentQuestionId);
-    return [_question, _answers];
-  }, [currentQuestionId]);
+  const configuration = useAppSelector(
+    (state) => state.quizSettings.configuration,
+  );
+
+  const questions = useAppSelector((state) => state.quizSettings.questions);
+
+  const fetchQuestions = async () => {
```

Comment:
I think this shouldn't be in a component
please find a better corresponding place to keep API calls